andela/travel_tool_front
# #161208596 FormsAPI Clean up & Documentation

#### What does this PR do?
- Adds a README to the FormsAPI for documentation
- Improves code in the FormsAPI and individual forms
  - Removes redundant chunks of code
  - Enables selection-based inputs like `dropdown` to have a `value` and `label` for each choice


#### Description of Task to be completed?
- Add a README to the FormsAPI for documentation and usage reference
- Improve the FormsAPI codebase
  - Remove unused pieces of code
  - Enable choice-based inputs to take strings or JavaScript objects for label and value

#### How should this be manually tested?
- Clone this repo - `git clone https://github.com/andela/travel_tool_front.git`
- Switch to the repo directory - `cd travel_tool_front`
- Check out this branch - `git checkout ch-forms-api-doc`
- This task is not expected to change any behaviour in the application but to make improvements to the codebase and ensure consistency in the implementation of the forms in the application
  - Checkout the Wiki section on this repo for the [documentation](https://github.com/andela/travel_tool_front/wiki/A-Guide-to-the-FormsAPI)
  - Pick a  form in the application and compare the code with the FormsAPI guide
  - You should be able to see how the pieces are coming together

#### Any background context you want to provide?
- The documentation is viewable on this repo's Wiki section

#### What are the relevant pivotal tracker stories?
[#161208596](https://www.pivotaltracker.com/story/show/161208596)

#### Screenshots (if appropriate)
N/A

#### Questions:
